IP set on wrong interface when running in a network namespace (linux)

When attempting to run in a network namespace, I came across an unusual bug:

```
$ ip netns add node1
$ ip link add veth0 type veth peer name veth1
$ ip link set veth0 up
$ ip link set veth1 netns node1 up
$ ip netns exec node1 su $USER
$ ip link set lo up
$ ./yggdrasil --autoconf
2018/04/26 20:03:17 Initializing...
2018/04/26 20:03:17 Starting interface...
2018/04/26 20:03:17 Started interface
2018/04/26 20:03:17 Starting admin socket...
2018/04/26 20:03:17 Started admin socket
2018/04/26 20:03:17 Starting TUN/TAP...
2018/04/26 20:03:17 Listening for TCP on: [::]:33859
2018/04/26 20:03:17 Listening for UDP on: [::]:44995
2018/04/26 20:03:17 Admin socket listening on [::1]:9001
2018/04/26 20:03:17 Started...
2018/04/26 20:03:19 Connected: fd01:1cc6:49a1:36a2:7706:7358:eb44:ec9@fe80::a41c:8fff:fe05:6072%veth1
# Then attempt to ping the node
panic: write /dev/net/tun: input/output error

goroutine 25 [running]:
yggdrasil.(*tunDevice).write(0xc4201a62e0, 0x0, 0x0)
	/home/arceliar/Misc/mesh/yggdrasil-go/src/yggdrasil/tun.go:59 +0x242
yggdrasil.(*Core).DEBUG_startTunWithMTU.func2(0xc4201a0000)
	/home/arceliar/Misc/mesh/yggdrasil-go/src/yggdrasil/debug.go:241 +0x33
created by yggdrasil.(*Core).DEBUG_startTunWithMTU
	/home/arceliar/Misc/mesh/yggdrasil-go/src/yggdrasil/debug.go:241 +0x146
```

On closer inspection, it appears that the tun device was never brought up, and the `fd00::/8` address within the `node1` namespace was assigned to `veth1`.

I'm guessing this happened as part of the update to use `github.com/docker/libcontainer/netlink` instead of calling `ip` from the command line to set addresses and bring interfaces up. Ideally, netns should work, given its usefulness in the past for testing, but the bare minimum requirement is to raise an error when the tun fails to go up and avoid setting the address on the wrong interface.

--- a/misc/run-schannel-netns
+++ b/misc/run-schannel-netns
@@ -44,6 +44,13 @@ ip netns exec node5 tc qdisc add dev veth54 root tbf rate 100mbit burst 8192 lat
 ip netns exec node4 tc qdisc add dev veth46 root tbf rate 10mbit burst 8192 latency 1ms
 ip netns exec node6 tc qdisc add dev veth64 root tbf rate 10mbit burst 8192 latency 1ms
 
+ip netns exec node1 ip link set lo up
+ip netns exec node2 ip link set lo up
+ip netns exec node3 ip link set lo up
+ip netns exec node4 ip link set lo up
+ip netns exec node5 ip link set lo up
+ip netns exec node6 ip link set lo up
+
 ip netns exec node1 ./run --autoconf --pprof &> /dev/null &
 ip netns exec node2 ./run --autoconf --pprof &> /dev/null &
 ip netns exec node3 ./run --autoconf --pprof &> /dev/null &

--- a/src/yggdrasil/debug.go
+++ b/src/yggdrasil/debug.go
@@ -236,6 +236,7 @@ func (c *Core) DEBUG_startTunWithMTU(ifname string, iftapmode bool, mtu int) {
 		if err != nil {
 			panic(err)
 		}
+		c.log.Println("Setup TUN/TAP:", c.tun.iface.Name(), straddr)
 		go func() { panic(c.tun.read()) }()
 	}
 	go func() { panic(c.tun.write()) }()

--- a/src/yggdrasil/tun_linux.go
+++ b/src/yggdrasil/tun_linux.go
@@ -48,7 +48,8 @@ func (tun *tunDevice) setupAddress(addr string) error {
 	}
 	for _, ifce := range ifces {
 		if ifce.Name == tun.iface.Name() {
-			netIF = &ifce
+			var newIF = ifce
+			netIF = &newIF // Don't point inside ifces, it's apparently unsafe?...
 		}
 	}
 	if netIF == nil {

